File: 2808-minimum-seconds-to-equalize-a-circular-array/2808-minimum-seconds-to-equalize-a-circular-array.java
class Solution {
    public int minimumSeconds(List<Integer> nums) {
        int n=nums.size();
        Map<Integer,List<Integer>>fre=new HashMap<>();
        for(int i=0;i<n;i++){
            int num = nums.get(i);
            if (!fre.containsKey(num)) {
                fre.put(num, new ArrayList<>());
            }
            fre.get(num).add(i);
        }
        int ans=n;
        for(Map.Entry<Integer,List<Integer>>entry:fre.entrySet()){
            int x=entry.getKey();
            int cans=0;
            List<Integer>positions=entry.getValue();
            for (int i = 0; i < positions.size() - 1; i++) {
                int diff = (positions.get(i + 1) - positions.get(i)) / 2;
                cans = Math.max(cans, diff);
            }

            cans = Math.max(cans, (positions.get(0) + (n - 1 -                         positions.get(positions.size() - 1) + 1)) / 2);
            ans = Math.min(ans, cans);
        }
        return ans;
        
    }
}

File: 2808-minimum-seconds-to-equalize-a-circular-array/2808-minimum-seconds-to-equalize-a-circular-array.cpp
class Solution {
public:
    int minimumSeconds(vector<int>& nums) {
        int n=nums.size();
        map<int,vector<int>>fre;
        for(int i=0;i<n;i++)fre[nums[i]].push_back(i);
        int ans=n;
        for(auto &p:fre){
            int x=p.first;
            int cans=0;
            for(int i=0;i<fre[x].size()-1;i++){
                int diff=(fre[x][i+1]-fre[x][i])/2;
                cans=max(cans,diff);
            }
            cans=max(cans,(fre[x][0]+(n-1-fre[x][fre[x].size()-1]+1))/2);
            ans=min(ans,cans);
        }
        return ans;
    }
};